src/rtp.cpp

```diff
@@ -130,12 +130,17 @@ void RtpExtensionHeader::setHeaderLength(uint16_t headerLength) {
 
 void RtpExtensionHeader::clearBody() { std::memset(getBody(), 0, getSize()); }
 
-void RtpExtensionHeader::writeCurrentVideoOrientation(size_t offset, uint8_t id, uint8_t value) {
-	if ((id == 0) || (id > 14) || ((offset + 2) > getSize()))
+void RtpExtensionHeader::writeOneByteHeader(size_t offset, uint8_t id, const byte *value, size_t size) {
+	if ((id == 0) || (id > 14) || ((offset + 1 + size) > getSize()))
 		return;
 	auto buf = getBody() + offset;
-	buf[0] = id << 4;
-	buf[1] = value;
+	buf[0] = id << 4 | (uint8_t(size) - 1);
```

Comment:
You need to check that `size > 0`.

Reply:
Fixed! I check that `size > 0` before writing length.

Also made sure to check size before copying.